src/lib/json-ld.js

```diff
@@ -0,0 +1,107 @@
+import { Helmet } from 'react-helmet';
+
+import config from '../../package.json';
+
+export function ArticleJsonLd({ post = {}, siteTitle = '' }) {
+  const { homepage = '', favicon = 'favicon.ico' } = config;
+  const { title, slug, excerpt, date, author, categories, modified, featuredImage } = post;
+
+  /** TODO - As image is a recommended field would be interesting to have a
+   * default image in case there is no featuredImage comming from WP,
+   * like the open graph social image
+   * */
+
+  const jsonLd = {
+    '@context': 'https://schema.org',
+    '@type': 'Article',
+    mainEntityOfPage: {
+      '@type': 'WebPage',
+      '@id': `${homepage}/posts/${slug}`,
+    },
+    headline: title,
+    image: [featuredImage?.sourceUrl],
+    datePublished: new Date(date).toISOString(),
+    dateModified: new Date(modified).toISOString() || new Date(date).toISOString(),
+    description: excerpt,
+    keywords: [categories.map(({ name }) => `${name}`).join(', ')],
+    copyrightYear: new Date(date).getFullYear(),
+    author: {
+      '@type': 'Person',
+      name: author?.name,
+    },
+    publisher: {
+      '@type': 'Organization',
+      name: siteTitle,
+      logo: {
+        '@type': 'ImageObject',
+        url: `${homepage}/${favicon}`,
+      },
+    },
+  };
+
+  return (
+    <Helmet encodeSpecialCharacters={false}>
+      <script type="application/ld+json">{JSON.stringify(jsonLd)}</script>
+    </Helmet>
+  );
+}
+
+export function WebsiteJsonLd({ siteTitle = '' }) {
+  const { homepage = '' } = config;
+
+  const jsonLd = {
+    '@context': 'https://schema.org',
+    '@type': 'WebSite',
+    name: siteTitle,
+    url: homepage,
+    copyrightYear: new Date().getFullYear(),
+    potentialAction: {
+      '@type': 'SearchAction',
+      target: `${homepage}/search/?q={search_term_string}`,
+      'query-input': 'required name=search_term_string',
+    },
+  };
+
+  return (
+    <Helmet encodeSpecialCharacters={false}>
+      <script type="application/ld+json">{JSON.stringify(jsonLd)}</script>
+    </Helmet>
+  );
+}
+
+export function WebpageJsonLd({ title = '', description = '', siteTitle = '' }) {
```

Comment:
if we have the `title` and the `description` here - woudl we also have the page's path that we could pass in to construct the `url`?

Reply:
I forgot `url` attribute, thanks for catching it! Will make the changes